ext/RastersProjExt/cellarea.jl

```diff
@@ -0,0 +1,185 @@
+    two_pi_R2 = 2 * pi * radius * radius
+    band_area = broadcast(DD.intervalbounds(lat)) do yb
+        two_pi_R2 * (sind(yb[2]) - sind(yb[1]))
+    end
+    
+    broadcast(DD.intervalbounds(lon), band_area') do xb, ba
+        abs((xb[2] - xb[1]) / 360 * ba)
+    end
+end
+
+function _spherical_cellarea(dims::Tuple{<:XDim, <:YDim}; radius = 6371008.8, use_area_of_use = true)
+    # check the dimensions 
+    isnothing(crs(dims)) && _no_crs_error()
+
+    areas = if _isdegrees(crs(dims)) # check if need to reproject
+        _area_from_lonlat(dims...; radius)
+    elseif !isnothing(mappedcrs(dims)) && _isdegrees(mappedcrs(dims))
+        _area_from_lonlat(reproject(dims; crs = mappedcrs(dims))...; radius)
+    else
+        xbnds, ybnds = DD.intervalbounds(dims)
+        R2 = radius * radius
+        area_of_use = if use_area_of_use
+            # Use a temporary transformation object to get the area of use,
+            # since the transformation object has to be recreated with the area of use
+            _get_area_of_use(Proj.Transformation(crs(dims), EPSG(4326); always_xy = true), Extents.Extent(X = extrema(dims[1]), Y = extrema(dims[2])))
+        else
+            C_NULL
+        end
+
+        transform = Proj.Transformation(crs(dims), EPSG(4326); always_xy = true, area = area_of_use)
+
+        result = [
+                _area_from_coords(
+                    transform,         
+                    (
+                        (xb[1], yb[1]), 
+                        (xb[2], yb[1]), 
+                        (xb[2], yb[2]), 
+                        (xb[1], yb[2]),
+                    )
+                ) * R2
+                for xb in xbnds, yb in ybnds
+            ]
+        
+        if use_area_of_use
+            Proj.proj_area_destroy(area_of_use)
+        end
+
+        return result
+    end
+end
+
+function _get_area_of_use(transform::Proj.Transformation, extent::Extents.Extent; densify_pts = 21)
+    # Transform the extent using `proj_trans_bounds`
+    (xmin, xmax), (ymin, ymax) = Proj.bounds(transform, extent.X, extent.Y; densify_pts)
+
+    # Create an area of use object 
+    # This MUST be destroyed by the caller
+    area = Proj.proj_area_create()
+
+    # Set the bounding box of the area of use
+    Proj.proj_area_set_bbox(
+        area, 
+        xmin, # west_lon_degree
+        ymin, # south_lat_degree
+        xmax, # east_lon_degree
+        ymax  # north_lat_degree
+    )
+
+    return area
+end
+
+# TODO: put these in Proj (specifically the dispatches on GFT types)
+_isgeographic(crs) = _isgeographic(Proj.CRS(crs))
+_isgeographic(crs::Proj.CRS) = Proj.is_geographic(crs)
+
+_isdegrees(crs) = _isdegrees(Proj.CRS(crs))
+function _isdegrees(crs::Proj.CRS)
+    _isgeographic(crs) || return false
+    # This is a tiny bit inefficient, but it takes 500ns on my machine, 
+    # so I think we can disregard the inefficiency...
+    return axis_is_degrees(crs, 0) && axis_is_degrees(crs, 1)
+end
```

Comment:
I think `Proj.is_geographic` only returns `true` for lon-lat systems (that's what it says in the pyproj documentation at least https://pyproj4.github.io/pyproj/stable/api/crs/crs.html#pyproj.crs.CRS.is_geographic)

Reply:
You're right, this is the GDAL implementation:

https://github.com/OSGeo/gdal/blob/60ab2831abf9446c629b3a417a16cb35e7f41e63/gdal/ogr/ogrspatialreference.cpp#L7671-L7703

so we need to implement that same munging here.  Can't say I'm very fond of this, wish GDAL just upstreamed that to Proj...

Reply:
Ah no, Proj already implements the GDAL correction!  So we should be fine.

Reply:
Do you have examples of CRS that we should test?